app/src/main/java/com/mybus/view/FavoritesCardView.java

```diff
@@ -31,28 +37,33 @@ public FavoritesCardView(Context context, AttributeSet attrs) {
 
     private void init() {
         inflate(getContext(), R.layout.favorites_card_view, this);
+        mRecyclerView = (RecyclerView) findViewById(R.id.favorites_list);
+        mNoFavoritesTextView = (TextView) findViewById(R.id.noFavorites);
+    }
 
-        mFavoriteItemsList = (RecyclerView) findViewById(R.id.favorites_list);
-
-        // use this setting to improve performance if you know that changes
-        // in content do not change the layout size of the RecyclerView
-        mFavoriteItemsList.setHasFixedSize(true);
+    public void setItemList(List<FavoriteLocation> list) {
+        mFavoriteLocationList = list;
+        if (mFavoriteLocationList != null && mFavoriteLocationList.size() > 0) {
+            mNoFavoritesTextView.setVisibility(GONE);
+            mRecyclerView.setVisibility(VISIBLE);
 
-        // use a linear layout manager
-        mLayoutManager = new LinearLayoutManager(getContext());
-        mFavoriteItemsList.setLayoutManager(mLayoutManager);
+            // use this setting to improve performance if you know that changes
+            // in content do not change the layout size of the RecyclerView
+            mRecyclerView.setHasFixedSize(true);
+            // use a linear layout manager
+            mLayoutManager = new LinearLayoutManager(getContext());
+            mRecyclerView.setLayoutManager(mLayoutManager);
 
-        // specify an adapter (see also next example)
-        String[] myDataSet = {"Casa", "Trabajo", "Gym", "Casa Novia", "Casa", "Trabajo", "Gym", "Casa Novia"};
-        mListAdapter = new FavoriteItemAdapter(myDataSet);
-        mListAdapter.setItemSelectedListener(this);
-        mFavoriteItemsList.setAdapter(mListAdapter);
+            mListAdapter = new FavoriteItemAdapter(mFavoriteLocationList);
+            mListAdapter.setItemSelectedListener(this);
+            mRecyclerView.setAdapter(mListAdapter);
+        }
```

Comment:
Should have the:
`else{
         mNoFavoritesTextView.setVisibility(VISIBLE);
         mRecyclerView.setVisibility(GONE);
}`